blog/_posts/2021-08-20-symbiosis-symbulation.md

```diff
@@ -0,0 +1,69 @@
+ 
+ For this summer, we focused on being able to add symbionts to a host system which would then increase the count of a point system. The original idea is that if the host reaches a certain amount of points, it would then donate a point to the symbiont and if the symbiont reaches a certain amount of points, then the symbiont would donate a point to the host which would then be multiplied by three.
+
+---
+
+ ## Symbiosis Environment
+ In order to create symbiosis, we need to create a host with symbionts and create a world to put them in. Header files for the world, host and symbiont were created in order to do so. These files were originally created by the last WAVES participant that worked with Anya, which were then built upon and edited for this summer. 
+ 
+The constructors for each organism were edited to be up-to-date, which additional functions to add symbionts, which would then increase the count when simulating. 
+
+For creating the world, a similar approach was taken for simulating a world in Symbulation. Since we didn't need all the functions and other attributes as the SymWorld in Symbulation, we inherited certain functions that were necessary for this project. For our world, an empirical world constructor was used to birth the organisms. In the world, two different selection methods were used; elite select and tournament select. Elite select picks a set of the most fit individuals for the next generation while tournament select randomly samples a subset of the population and then chooses the most-fit individual in that sample to progress to the next generation. These functions were originally created in Empirical, but changed slightly to fit this project, which is shown for tournament select below.
+
+```
+void AvidaGPTournamentSelect(AvidaGPWorld & world, size_t t_size, size_t tourny_count=1) {
+    emp_assert(t_size > 0, "Cannot have a tournament with zero organisms.", t_size, world.GetNumOrgs());
+    emp_assert(t_size <= world.GetNumOrgs(), "Tournament too big for world.", t_size, world.GetNumOrgs());
+    emp_assert(tourny_count > 0);
+
+    emp::vector<size_t> entries;
+    for (size_t T = 0; T < tourny_count; T++) {
+      entries.resize(0);
+      // Choose organisms for this tournament (with replacement!)
+      for (size_t i=0; i < t_size; i++) entries.push_back(
+        world.GetRandomOrgID()
+      );
+
+      double best_fit = world.CalcFitnessID(entries[0]);
+      size_t best_id = entries[0];
+
+      // Search for a higher fit org in the tournament.
+      for (size_t i = 1; i < t_size; i++) {
+        const double cur_fit = world.CalcFitnessID(entries[i]);
+        if (cur_fit > best_fit) {
+          best_fit = cur_fit;
+          best_id = entries[i];
+        }
+      }
+
+      // Place the highest fitness into the next generation!
```

Comment:
It'd be nice to have a note of what you did have to change, ie passing the pointer to DoBirth and DoBirth was where you spent a fair amount of time, so maybe discussion of it should be included as well.